examples/app/headless_renderer.rs

```diff
@@ -0,0 +1,508 @@
+            render_target_image.resize(size);
+            let render_target_image_handle = images.add(render_target_image);
+
+            // This is the texture that will be copied to.
+            let mut cpu_image = Image {
+                texture_descriptor: TextureDescriptor {
+                    label: None,
+                    size,
+                    dimension: TextureDimension::D2,
+                    format: TextureFormat::Rgba8UnormSrgb,
+                    mip_level_count: 1,
+                    sample_count: 1,
+                    usage: TextureUsages::COPY_DST | TextureUsages::TEXTURE_BINDING,
+                    view_formats: &[],
+                },
+                ..Default::default()
+            };
+            cpu_image.resize(size);
+            let cpu_image_handle = images.add(cpu_image);
+
+            commands.spawn(ImageCopier::new(
+                render_target_image_handle.clone(),
+                size,
+                render_device,
+            ));
+
+            commands.spawn(ImageToSave(cpu_image_handle));
+
+            scene_controller.state = SceneState::Render(pre_roll_frames);
+            scene_controller.name = scene_name;
+            RenderTarget::Image(render_target_image_handle)
+        }
+
+        // Takes from channel image content sent from render world and saves it to disk
+        fn update(
+            images_to_save: Query<&ImageToSave>,
+            receiver: Res<MainWorldReceiver>,
+            mut images: ResMut<Assets<Image>>,
+            mut scene_controller: ResMut<SceneController>,
+            mut app_exit_writer: EventWriter<AppExit>,
+        ) {
+            if let SceneState::Render(n) = scene_controller.state {
+                if n < 1 {
+                    use rand::Rng;
+                    let mut rng = rand::thread_rng();
+
+                    // We don't want to block the main world on this,
+                    // so we use try_recv which attempts to receive without blocking
+                    while let Ok(data) = receiver.try_recv() {
+                        for image in images_to_save.iter() {
+                            let img_bytes = images.get_mut(image.id()).unwrap();
+                            img_bytes.data = data.clone();
+
+                            let img = match img_bytes.clone().try_into_dynamic() {
+                                Ok(img) => img.to_rgba8(),
+                                Err(e) => panic!("Failed to create image buffer {e:?}"),
+                            };
+
+                            let images_dir =
+                                PathBuf::from(env!("CARGO_MANIFEST_DIR")).join("test_images");
+                            println!("Saving image to: {images_dir:?}");
+                            std::fs::create_dir_all(&images_dir).unwrap();
+
+                            let number = rng.gen::<u128>();
+                            let image_path = images_dir.join(format!("{number:032X}.png"));
```

Comment:
Could you use an incrementing number instead of a random value?

Reply:
I used random value to prevent filenames from matching when somebody rerun example. I suggest using datetime stamps as more descriptive option.